### ISSUE TITLE
Error in ov.pl.plot_flowsig_network

### ISSUE THREAD

**Issue by YimingLiu000** [Starlitnightly/omicverse#304]
**Describe the bug**
When running the spatial communication example tutorial, an error occurs in the final visualization function

**To Reproduce**

![Image](https://github.com/user-attachments/assets/4d0bdf8a-7984-4d26-b252-d3ebe1e0d060)

**Expected behavior**

![Image](https://github.com/user-attachments/assets/49cb0c45-9dbb-40eb-b68a-fbbe89ffe14d)

**Additional context**
Package                       Version          Build
----------------------------- ---------------- -----
absl-py                       2.1.0
adjustText                    1.3.0
aiobotocore                   2.5.4
aiohappyeyeballs              2.4.6
aiohttp                       3.11.12
aioitertools                  0.12.0
aiosignal                     1.3.2
alabaster                     0.7.16
anndata                       0.10.8
anndata2ri                    1.3.2
annotated-types               0.7.0
annoy                         1.17.3
anyio                         4.3.0
appdirs                       1.4.4
arboreto                      0.1.6
argon2-cffi                   23.1.0
argon2-cffi-bindings          21.2.0
array_api_compat              1.6
arrow                         1.3.0
arviz                         0.17.1
asciitree                     0.3.3
astor                         0.8.1
asttokens                     2.4.1
async-lru                     2.0.4
async-timeout                 5.0.1
attrs                         23.1.0
autograd                      1.6.2
autograd-gamma                0.5.0
Babel                         2.14.0
bbknn                         1.6.0
beautifulsoup4                4.12.3
biopython                     1.83
bioservices                   1.11.2
biothings-client              0.3.0
bleach                        6.1.0
bokeh                         3.4.3
boltons                       24.0.0
botocore                      1.31.17
Brotli                        1.0.9
cached-property               1.5.2
cachetools                    5.5.0
CAST                          0.4
cattrs                        23.2.3
causaldag                     0.1a163
cellphonedb                   5.0.1
cellrank                      2.0.6
celltypist                    1.6.3
certifi                       2025.4.26
cffi                          1.16.0
cfgv                          3.4.0
cftime                        1.6.4.post1
chardet                       5.2.0
charset-normalizer            2.0.4
chex                          0.1.7
click                         8.1.8
cloudpickle                   1.5.0
colorcet                      3.1.0
colorlog                      6.8.2
comm                          0.2.2
commonmark                    0.9.1
conditional-independence      0.1a6
contextlib2                   21.6.0
contourpy                     1.2.1
cosg                          1.0.1
ctxcore                       0.2.0
cucim-cu12                    24.4.0
cuda-python                   12.6.0
cudf-cu12                     24.4.1
cugraph-cu12                  24.4.0
cuml-cu12                     24.4.0
cuproj-cu12                   24.4.0
cupy-cuda12x                  13.3.0
cuspatial-cu12                24.4.0
custom-inherit                2.4.1
cuvs-cu12                     24.4.0
cuxfilter-cu12                24.4.1
cvxopt                        1.3.0
cycler                        0.12.1
Cython                        3.0.10
cytoolz                       1.0.1
dask                          2024.8.0
dask-cuda                     24.4.0
dask-cudf-cu12                24.4.1
dask-expr                     1.1.10
dask-image                    2024.5.3
dataclasses                   0.6
datashader                    0.16.0
DateTime                      5.5
debugpy                       1.8.1
decorator                     5.1.1
decoupler                     1.5.0
deeptime                      0.4.5
defusedxml                    0.7.1
Deprecated                    1.2.14
dgl                           2.4.0+cu121
dill                          0.4.0
distributed                   2024.8.0
distributed-ucxx-cu12         0.37.0
dm-tree                       0.1.8
docopt                        0.6.2
docrep                        0.3.2
docutils                      0.21.2
drawsvg                       2.4.0
easydev                       0.13.1
einops                        0.7.0
entrypoints                   0.4
equinox                       0.11.8
ERgene                        1.2.8
et_xmlfile                    2.0.0
ete3                          3.1.3
etils                         1.5.2
exceptiongroup                1.2.0
executing                     2.0.1
fasteners                     0.19
fastjsonschema                2.19.1
fastrlock                     0.8.2
fbpca                         1.0
filelock                      3.17.0
flax                          0.8.2
fonttools                     4.51.0
formulaic                     1.0.1
fqdn                          1.5.1
frozendict                    2.4.2
frozenlist                    1.5.0
fsspec                        2023.6.0
ftpretty                      0.4.0
future                        1.0.0
gdown                         5.2.0
geopandas                     1.0.1
geosketch                     1.2
get-annotations               0.1.2
gevent                        24.2.1
goatools                      1.4.12
google-ai-generativelanguage  0.6.15
google-api-core               2.25.1
google-api-python-client      2.172.0
google-auth                   2.40.3
google-auth-httplib2          0.2.0
google-generativeai           0.8.5
googleapis-common-protos      1.70.0
graphical-model-learning      0.1a8
graphical-models              0.1a21
graphlib-backport             1.0.3
graphtools                    1.5.3
greenlet                      3.0.3
grequests                     0.7.0
grpcio                        1.73.0
grpcio-status                 1.71.0
gseapy                        0.10.8
gtfparse                      2.5.0
gudhi                         3.11.0
h11                           0.14.0
h2                            4.1.0
h5netcdf                      1.4.1
h5py                          3.8.0
h5sparse                      0.1.0
harmonypy                     0.0.9
hnswlib                       0.7.0
holoviews                     1.19.1
hpack                         4.0.0
httpcore                      1.0.5
httplib2                      0.22.0
httpx                         0.27.0
huggingface-hub               0.25.2
hyperframe                    6.0.1
identify                      2.6.3
idna                          3.4
igraph                        0.11.8
imageio                       2.37.0
imagesize                     1.4.1
importlib_metadata            7.1.0
importlib_resources           6.4.0
infercnvpy                    0.4.5
inflect                       7.4.0
iniconfig                     2.0.0
interface-meta                1.3.0
interlap                      0.2.7
intervaltree                  3.1.0
ipdb                          0.13.13
ipykernel                     6.29.3
ipython                       8.18.1
ipywidgets                    8.1.2
isoduration                   20.11.0
jax                           0.4.23
jax-cuda12-pjrt               0.4.23
jax-cuda12-plugin             0.4.23
jaxlib                        0.4.23
jaxopt                        0.8.3
jaxtyping                     0.2.34
jedi                          0.19.1
Jinja2                        3.0.3
jmespath                      1.0.1
joblib                        1.3.2
json5                         0.9.25
jsonpointer                   2.4
jsonschema                    4.21.1
jsonschema-specifications     2023.12.1
jupyter                       1.1.1
jupyter_client                8.6.1
jupyter-console               6.6.3
jupyter_core                  5.7.2
jupyter-events                0.10.0
jupyter-lsp                   2.2.5
jupyter_server                2.14.0
jupyter_server_proxy          4.4.0
jupyter_server_terminals      0.5.3
jupyterlab                    4.4.3
jupyterlab_pygments           0.3.0
jupyterlab_server             2.27.3
jupyterlab_widgets            3.0.10
kiwisolver                    1.4.5
ktplotspy                     0.2.3
lazy_loader                   0.4
legacy-api-wrap               1.4
leidenalg                     0.10.2
liana                         1.4.0
libpysal                      4.8.1
libucx-cu12                   1.15.0.post1
lifelines                     0.27.8
lightning                     2.1.4
lightning-utilities           0.12.0
line-profiler                 4.1.2
lineax                        0.0.5
linkify-it-py                 2.0.3
llvmlite                      0.42.0
locket                        1.0.0
loguru                        0.7.2
loompy                        3.0.7
louvain                       0.8.2
lxml                          5.2.1
lz4                           4.3.3
Markdown                      3.6
markdown-it-py                3.0.0
MarkupSafe                    2.1.3
matplotlib                    3.6.3
matplotlib-inline             0.1.7
matplotlib-scalebar           0.9.0
matplotlib-venn               1.1.1
mdit-py-plugins               0.4.2
mdtraj                        1.10.3
mdurl                         0.1.2
mellon                        1.4.2
metatime                      1.3.0
mistune                       3.0.2
mizani                        0.9.3
mkl-fft                       1.3.8
mkl-random                    1.2.4
mkl-service                   2.4.0
ml-collections                0.1.1
ml-dtypes                     0.4.0
mofax                         0.3.6
more-itertools                10.7.0
mpi4py                        4.0.3
mpmath                        1.3.0
msgpack                       1.1.0
mudata                        0.2.4
multidict                     6.1.0
multipledispatch              0.6.0
multiprocess                  0.70.18
multiprocessing-on-dill       3.5.0a4
multiscale_spatial_image      1.0.1
munkres                       1.1.4
muon                          0.1.6
mygene                        3.2.2
natsort                       8.4.0
nbclient                      0.10.0
nbconvert                     7.16.3
nbformat                      5.10.4
nest_asyncio                  1.6.0
netCDF4                       1.7.2
networkx                      2.8
nodeenv                       1.9.1
notebook                      7.4.3
notebook_shim                 0.2.4
numba                         0.59.0
numcodecs                     0.12.1
numexpr                       2.8.4
numpy                         1.23.5
numpy-groupies                0.11.2
numpyro                       0.15.0
nvidia-cublas-cu12            12.1.3.1
nvidia-cuda-cupti-cu12        12.1.105
nvidia-cuda-nvcc-cu12         12.6.77
nvidia-cuda-nvrtc-cu12        12.1.105
nvidia-cuda-runtime-cu12      12.1.105
nvidia-cudnn-cu12             9.1.0.70
nvidia-cufft-cu12             11.0.2.54
nvidia-curand-cu12            10.3.2.106
nvidia-cusolver-cu12          11.4.5.107
nvidia-cusparse-cu12          12.1.0.106
nvidia-cusparselt-cu12        0.6.2
nvidia-nccl-cu12              2.20.5
nvidia-nvjitlink-cu12         12.4.127
nvidia-nvtx-cu12              12.1.105
nvtx                          0.2.10
OMB                           0.1.dev0
ome-zarr                      0.10.2
omicverse                     1.7.1
omnipath                      1.0.8
openchord                     0.1.7
opencv-contrib-python         4.11.0.86
openpyxl                      3.1.5
opt_einsum                    3.4.0
optax                         0.2.1
orbax-checkpoint              0.5.9
ott-jax                       0.4.9
overrides                     7.7.0
packaging                     24.0
palettable                    3.3.3
pandas                        2.0.3
pandocfilters                 1.5.0
panel                         1.4.5
param                         2.1.0
parso                         0.8.4
partd                         1.4.1
pathos                        0.3.4
patsy                         0.5.6
pcurvepy2                     0.1.5
pertpy                        0.6.0
petsc                         3.23.3
petsc4py                      3.23.3
pexpect                       4.9.0
pgmpy                         0.1.24
phate                         1.0.11
pickleshare                   0.7.5
pillow                        10.2.0
PIMS                          0.7
pip                           25.1.1
pipdeptree                    2.18.1
pkgutil_resolve_name          1.3.10
platformdirs                  4.2.0
plotly                        5.24.1
plotnine                      0.12.4
plottable                     0.1.5
pluggy                        1.5.0
ply                           3.11
polars                        0.20.31
pooch                         1.8.2
POT                           0.4.0
pox                           0.3.6
ppft                          1.7.7
pre_commit                    4.0.1
progressbar2                  4.4.2
prometheus_client             0.20.0
prompt-toolkit                3.0.42
propcache                     0.3.0
proto-plus                    1.26.1
protobuf                      5.29.5
psutil                        5.9.0
ptyprocess                    0.7.0
pure-eval                     0.2.2
py-pcha                       0.1.3
pyarrow                       14.0.2
pyarrow-hotfix                0.6
pyasn1                        0.6.1
pyasn1_modules                0.4.2
pybind11                      2.12.0
pycairo                       1.27.0
PyComplexHeatmap              1.8.2
pycparser                     2.22
pyct                          0.5.0
pydantic                      2.11.5
pydantic_core                 2.33.2
pydeseq2                      0.4.12
pydot                         3.0.2
pyEMMA                        2.5.12
pygam                         0.8.0
Pygments                      2.17.2
pygpcca                       1.0.4
pygraphviz                    1.11
PyGSP                         0.5.1
pylibcugraph-cu12             24.4.0
pylibraft-cu12                24.4.0
pyliger                       0.2.4
pymde                         0.1.18
pynndescent                   0.5.12
pynvjitlink-cu12              0.3.0
pynvml                        11.4.1
pyogrio                       0.10.0
Pyomo                         6.8.1
pyparsing                     3.0.9
pypng                         0.20220715.0
pyproj                        3.6.1
PyQt5                         5.15.11
PyQt5-Qt5                     5.15.15
PyQt5_sip                     12.15.0
pyreadr                       0.5.3
pyro-api                      0.1.2
pyro-ppl                      1.9.1
pysam                         0.22.0
pyscenic                      0.12.1
PySocks                       1.7.1
pytest                        8.3.4
python-circos                 0.3.0
python-dateutil               2.9.0
python-dotplot                0.0.1
python-igraph                 0.11.8
python-json-logger            2.0.7
python-utils                  3.8.2
pytoml                        0.1.21
pytorch-lightning             2.5.0.post0
pytz                          2024.1
pyviz_comms                   3.0.3
PyYAML                        6.0.1
pyzmq                         26.0.0
raft-dask-cu12                24.4.0
rapids-dask-dependency        24.4.1
rapids_singlecell             0.10.10
referencing                   0.34.0
regex                         2024.9.11
reportlab                     4.2.5
requests                      2.31.0
requests-cache                1.2.0
rfc3339-validator             0.1.4
rfc3986-validator             0.1.1
rich                          13.9.4
rmm-cu12                      24.4.0
rpds-py                       0.18.0
rpy2                          3.5.16
rsa                           4.9.1
rtree                         1.4.0
s-gd2                         1.8.1
s3fs                          2023.6.0
safetensors                   0.4.5
scanorama                     1.7.4
scanpy                        1.10.1
scgen                         2.1.1
schist                        0.8.4
scib                          1.1.5            1
scib-metrics                  0.5.1
scikit-dimension              0.3.4
scikit-image                  0.22.0
scikit-learn                  1.6.0
scikit-misc                   0.3.1
scipy                         1.11.4
scprep                        1.2.3
scrublet                      0.2.3
scvelo                        0.3.3
scvi-colab                    0.12.0
scvi-tools                    1.1.2
seaborn                       0.13.0
Send2Trash                    1.8.3
session-info                  1.0.0
setuptools                    75.8.0
shapely                       2.0.6
simpervisor                   1.0.0
six                           1.16.0
slepc                         3.23.1
slepc4py                      3.23.1
slicerator                    1.1.0
sniffio                       1.3.1
snowballstemmer               2.2.0
sortedcontainers              2.4.0
soupsieve                     2.5
sparsecca                     0.3.1
spatial_image                 1.1.0
spatialdata                   0.2.5.post0
Sphinx                        7.4.7
sphinx-autodoc-typehints      2.3.0
sphinxcontrib-applehelp       2.0.0
sphinxcontrib-devhelp         2.0.0
sphinxcontrib-htmlhelp        2.1.0
sphinxcontrib-jsmath          1.0.1
sphinxcontrib-qthelp          2.0.0
sphinxcontrib-serializinghtml 2.0.0
squidpy                       1.6.1
stack-data                    0.6.2
statannotations               0.4.0
statsmodels                   0.14.1
stdlib-list                   0.10.0
suds-community                1.1.2
switchlang                    0.1.0
sympy                         1.13.1
tangram-sc                    1.0.4
tasklogger                    1.2.0
tbb                           2022.0.0
tblib                         3.0.0
tcmlib                        1.2.0
tenacity                      9.0.0
tensorboard                   2.16.2
tensorboard-data-server       0.7.0
tensorstore                   0.1.56
termcolor                     2.4.0
terminado                     0.18.1
texttable                     1.7.0
threadpoolctl                 3.1.0
tifffile                      2024.8.30
tinycss2                      1.2.1
tokenizers                    0.20.1
tomli                         2.0.1
toolz                         0.12.1
torch                         2.4.0
torch-geometric               2.6.1
torch_scatter                 2.1.2+pt24cu121
torch_sparse                  0.6.18+pt24cu121
torchaudio                    2.5.1
torchmetrics                  1.6.1
torchvision                   0.20.1
tornado                       6.4
toyplot                       2.0.0
toytree                       2.0.5
tqdm                          4.65.0
traitlets                     5.14.2
transformers                  4.45.2
treelite                      4.1.2
triton                        3.0.0            1
typeguard                     2.13.3
types-python-dateutil         2.9.0.20240316
typing                        3.7.4.3
typing_extensions             4.14.0
typing-inspection             0.4.1
typing-utils                  0.1.0
tzdata                        2024.2
tzlocal                       5.2
uc-micro-py                   1.0.3
ucx-py-cu12                   0.37.0
ucxx-cu12                     0.37.0
umap-learn                    0.5.5
unicodedata2                  15.1.0
UpSetPlot                     0.9.0
uri-template                  1.3.0
uritemplate                   4.2.0
url-normalize                 1.4.3
urllib3                       1.26.20
validators                    0.35.0
velovi                        0.3.1
wcwidth                       0.2.13
webcolors                     1.13
webencodings                  0.5.1
websocket-client              1.7.0
Werkzeug                      3.0.2
wheel                         0.45.1
widgetsnbextension            4.0.10
wrapt                         1.16.0
xarray                        2024.3.0
xarray-dataclasses            1.9.1
xarray-datatree               0.0.15
xarray-einstats               0.7.0
xarray-schema                 0.0.3
xarray-spatial                0.4.0
xgboost                       2.1.4
XlsxWriter                    3.2.2
xmltodict                     0.13.0
xyzservices                   2024.9.0
yarl                          1.18.3
zarr                          2.18.2
zict                          3.0.0
zipp                          3.17.0
zope.event                    5.0
zope.interface                6.3
zstandard                     0.22.0
